src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -307,6 +325,49 @@ private void nullifyWatchdog() {
             watchdog = null;
         }
     }
+
+    /**
+     * Offers lines in buffer to queue.
+     *
+     * @param cb a buffer to have received text data.
+     * @return the line string. null if no EOL, otherwise buffer is compacted.
+     */
+    private String getLine(CharBuffer cb) {
+        String line = null;
+        int pos = cb.position();
+        cb.flip();
+        for (int i = 0; i < cb.length(); i++) {
+            if (cb.charAt(i) == '\n') {
+                line = getSubSequence(cb, 0, i).toString().trim();
+                cb.position(i + 1);
```

Comment:
Oh, I'm misunderstanding. EOL is either CRLF(Win) or LF(Linux, MacOS10). So buffer splitting is done by LF. CR in duplicated string will be stripped by String.trim(). No operation regarding CR handling on buffer.